lib/http/http.c

```diff
@@ -508,6 +509,16 @@ gotheaders(struct http_cookie * H, uint8_t * buf, size_t buflen)
 		H->res.headers[i].header = s;
 		H->res.headers[i].value = s[cpos] ? &s[cpos + 1] : &s[cpos];
 		s[cpos] = '\0';
+
+		/* Strip leading optional whitespace (OWS) from the value. */
+		H->res.headers[i].value += strspn(H->res.headers[i].value,
+		    " \t");
+
+		/* Strip trailing optional whitespace (OWS) from the value. */
```

Comment:
I'd be inclined to move this bit above the header/value splitting:

```
/* Remove trailing whitespace. */
while (s_len) {
    if ((s[s_len - 1] == ' ') || (s[s_len - 1] == '\t'))
        s[--s_len] = '\0';
}
```

No need to introduce a new variable `j` when we're inherently reducing the length of a string.

(Also, we need to check that we don't run past the *start* of the buffer!)

Reply:
LOL, infinite loop is infinite.